### DISCUSSION TITLE
Make running auto-off timer visible in UI

### DISCUSSION THREAD

**Discussion by freddair** [mongoose-os-apps/shelly-homekit#1469] [Ideas]
Let's expose the countdown timer visible in the UI beneath each switch auto-off setting.

Let's also disable the update button with a note if auto-off is running (thanks @timoschilling ), because device won't accept update if it's in "doing something" mode, including auto-off timer (thanks @rojer ).

https://github.com/mongoose-os-apps/shelly-homekit/issues/437

**Comment by rojer** [comment]
SGTM

**Comment by rojer** [comment]
real solution for the auto-off timer is to persist its state and restore on reboot so it continues running, but for now disabling update will do.

**Comment by andyblac** [comment]
@ZeevoX or @timoschilling could one of you guys take this on ?

**Comment by andyblac** [comment]
and maybe this could be done at the same time ?  https://github.com/mongoose-os-apps/shelly-homekit/issues/209

**Comment by timoschilling** [comment]
I can take a look on this issue, but #209 sound like a different feature.
